i2c/ina228_i2c/ina228_i2c.c

```diff
@@ -0,0 +1,119 @@
+    // Some registers are 2 bytes, some are 3 bytes, the accumilation registers are 5 bytes
+    uint8_t buffer_2[2];
+    uint8_t buffer_3[3];
+    uint8_t buffer_5[5];
+
+    // For some of the registers the 4 least significant bits are reserved and always read zero, which is why we shift right 12, 4, then left 4 for vshunt, vbus and current
+    i2c_write_blocking(i2c_default, I2C_ADDR, &VSHUNT_REG, 1, true);
+    i2c_read_blocking(i2c_default, I2C_ADDR, buffer_3, 3, false);
+    *vshunt = ((buffer_3[0] << 12) | (buffer_3[1] << 4) | (buffer_3[2] >> 4)) * VSHUNT_FACTOR;
+
+    i2c_write_blocking(i2c_default, I2C_ADDR, &VBUS_REG, 1, true);
+    i2c_read_blocking(i2c_default, I2C_ADDR, buffer_3, 3, false);
+    *vbus = ((buffer_3[0] << 12) | (buffer_3[1] << 4) | (buffer_3[2] >> 4)) * VBUS_FACTOR;
+
+    i2c_write_blocking(i2c_default, I2C_ADDR, &DIETEMP_REG, 1, true);
+    i2c_read_blocking(i2c_default, I2C_ADDR, buffer_2, 2, false);
+    *dietemp = ((buffer_2[0] << 8) + buffer_2[1]) * DIETEMP_FACTOR;
+
+    i2c_write_blocking(i2c_default, I2C_ADDR, &CURRENT_REG, 1, true);
+    i2c_read_blocking(i2c_default, I2C_ADDR, buffer_3, 3, false);
+    *current = ((buffer_3[0] << 12) | (buffer_3[1] << 4) | (buffer_3[2] >> 4)) * CURRENT_FACTOR;
+
+    i2c_write_blocking(i2c_default, I2C_ADDR, &POWER_REG, 1, true);
+    i2c_read_blocking(i2c_default, I2C_ADDR, buffer_3, 3, false);
+    *power = ((buffer_3[0] << 16) | (buffer_3[1] << 8) | buffer_3[2]) * POWER_FACTOR;
+
+    i2c_write_blocking(i2c_default, I2C_ADDR, &ENERGY_REG, 1, true);
+    i2c_read_blocking(i2c_default, I2C_ADDR, buffer_5, 5, false);
+    *energy = (((uint64_t) buffer_5[0] << 32) | (buffer_5[1] << 24) | (buffer_5[2] << 16) | (buffer_5[3] << 8) | buffer_5[4]) * ENERGY_FACTOR;
+
+    i2c_write_blocking(i2c_default, I2C_ADDR, &CHARGE_REG, 1, true);
+    i2c_read_blocking(i2c_default, I2C_ADDR, buffer_5, 5, false);
+    *charge = (((uint64_t) buffer_5[0] << 32) | (buffer_5[1] << 24) | (buffer_5[2] << 16) | (buffer_5[3] << 8) | buffer_5[4]) * CHARGE_FACTOR;
+}
+
+int main()
+{
+    stdio_init_all();
+
+    // Initialise external LED and turn it on (so we can measure some current)
+    gpio_init(EXT_LED_GPIO);
+    gpio_set_dir(EXT_LED_GPIO, GPIO_OUT);
+    gpio_put(EXT_LED_GPIO, true);
+
+    // I2C initialisation
+    i2c_init(i2c_default, 400*1000);
+    
+    // GPIO initialisiation
+    gpio_set_function(PICO_DEFAULT_I2C_SDA_PIN, GPIO_FUNC_I2C);
+    gpio_set_function(PICO_DEFAULT_I2C_SCL_PIN, GPIO_FUNC_I2C);
+    gpio_pull_up(PICO_DEFAULT_I2C_SDA_PIN);
+    gpio_pull_up(PICO_DEFAULT_I2C_SCL_PIN);
+
+    // Initialise ina228
+    ina228_init();
+
+    while (true) {
+        ina228_read(&vshunt, &vbus, &dietemp, &current, &power, &energy, &charge);
+        printf("INA228 Measurments:\nVSHUNT: %f\nVBUS: %f\nDIETEMP: %f\nCURRENT: %f\nPOWER: %f\nENERGY: %.9f\nCHARGE: %.9f\n-----------------\n", vshunt, vbus, dietemp, current, power, energy, charge);
+        sleep_ms(3000);
```

Comment:
3 seconds seems like a big sleep time, I think most of the other examples (that do things in a loop) sleep for one second?